remove strict check for branch in curated json

technophile-04 opened the pull request:
### Description: 

For CLI to work internally we don't need `branch` strict check. Also removed the strict check for descriptionl.

technophile-04 commented:
Ahh I see @damianmarti what you were saying because we had this in : 

https://github.com/scaffold-eth/create-eth/blob/b1fd6d40074e2ca35f6f7039fce94aefe67b0b76/src/curated-extensions.ts#L9

and branch could be optional you were telling us we default to `name` right? 

I think maybe we should add a new field `extensionFlag` because in above line `[ext.branch]`  is what CLI tracks when when user passes extension with `-e` we were assuming that it would be equal to `branch` name.  


So now difference b/w `name` and `extensionFlag` is `name` could be human readable name(only useful for scaffoldeth.io) and `extensionFlag` is codename thing which user passes as `-e` flag to CLI. If extension doesn't have `extensionFlag` by default it it's extension flag is `branch` name. 

```typescript
// json file type
type Extension = {
  extensionFlag?: string; // optional if not present we use branch
  branch?: string; // optional if not present we use extensionFlag
  repository: string;
  // use full for scaffoldeth.io
  description: string;
  version?: string; // (optional) if not present we default latest
  name?: string; // (optional) if not present we use the branch name as default (name might be useful to nicely display challenges names)
  // github: can be inferred by joining repositry + branch (scaffoldeth.io is requiring this for displaying link)
  // installCommand can inferred using repository + branch + version
}[];

```

rin-st commented:
> Ahh I see @damianmarti what you were saying because we had this in :

Wanted to ask the same thing)

What do you think about making `extensionFlag` required field? So it will be explicit and we won't need to suppose is it branch or not

Also thinking about better name but cant find something better for now). `extensionFlagValue`, `extensionCliValue`, `cliValue`? Make this field `name`, and rename current `name` to `title`?

damianmarti commented:
> > Ahh I see @damianmarti what you were saying because we had this in :
> 
> Wanted to ask the same thing)
> 
> What do you think about making `extensionFlag` required field for curated extensions? So it will be explicit and we won't need to suppose is it branch or not
> 
> Also thinking about better name but cant find something better for now). `extensionFlagValue`, `extensionCliValue`, `cliValue`? Make this field `name`, and rename current `name` to `title`?

maybe shortName?

technophile-04 commented:
> What do you think about making extensionFlag required field for curated extensions?

Yess makes sense! Updated it at https://github.com/scaffold-eth/create-eth/pull/161/commits/9fbe0524d4bb3caa3c8b1de80ebdf0aef94672ea and also changed the field name to `extensionFlagValue`. 


> Make this field name, and rename current name to title?
>
> maybe shortName?

The data submitted by builders in buidlguidl app has `name` field in json which correspondence to Card title in scaffoldeth.io . That's the reason I don't want to conflict it in the curated extension json which we will be sending to scaffoldeth.io

`name` => Something human-readable and useful for displayability on UI. 

`extensionFlagValue` => code name, something which people will pass as `-e` flag to CLI.

rin-st commented on src/curated-extensions.ts:
```diff
@@ -1,12 +1,27 @@
+    throw new Error(`Extension must have 'repository': ${JSON.stringify(ext)}`);
+  }
+  if (!ext.extensionFlagValue) {
+    throw new Error(`Extension must have 'extensionFlag': ${JSON.stringify(ext)}`);
```
`extensionFlagValue` here too

rin-st replied:
[3b049f5](https://github.com/scaffold-eth/create-eth/pull/161/commits/3b049f5009d5b8a50a544263ddb0f8eaf7ba4a04)

rin-st reviewed (approved):
Lgtm!
